### ISSUE TITLE
Warning: Read after posting article: 240 Message received.

### ISSUE THREAD

**Issue by ERR1R** [demanuel/NewsUP#42]
Just installed newsup.. it was a pain with perl version, cpan IO::Socket::SSL etc... lol.. anyways, i get this:

Warning: Read after posting article: 240 Message received.
Warning: Read after posting article: 240 Message received.
Warning: Read after posting article: 240 Message received.
Warning: Read after posting article: 240 Message received.
Warning: Read after posting article: 240 Message received.
Warning: Read after posting article: 240 Message received.
Warning: Read after posting article: 240 Message received.
Warning: Read after posting article: 240 Message received.
Warning: Read after posting article: 240 Message received.
Warning: Read after posting article: 240 Message received.
Warning: Read after posting article: 240 Message received.
Warning: Read after posting article: 240 Message received.

what does it mean and is it posting or not?

**Comment by ERR1R** [comment]
Ok, i just saw it posted to usenet.. why the warning?

http://www.newsdemon.com/Usenet-Server-Response-Codes.php

**Comment by demanuel** [comment]
Hi!

The files are being posted, but since the server doesn't return any message id, it will be unable to create a nzb file.

I changed the behaviour recently. Initially i was generating the IDs before starting uploading, but i stop doing that, because it's less stuff you write to the socket so the speeds will be better, and there were some concerns in the past about that leaking info: https://github.com/demanuel/NewsUP/issues/38

So for now, please use the version:
https://github.com/demanuel/NewsUP/tree/8659ae2ff8188036f5fe08716e7fac10c982df1d

Uploading is the same, the only difference, is that it won't have any progress indicator, besides the percentage.

**Comment by demanuel** [comment]
BTW, can you tell me your difficulties, so i can update the wiki?

you can send me an email, if you want.

**Comment by ERR1R** [comment]
Ok, is it possible to write the stuff on one line instead of filling my terminal? 

About the difficulties, i use debian and the wrong version of perl. So i needed to update perl and compile from source. I hope i can remember all what i had to do...

Install perl from source:
**_wget http://www.cpan.org/src/5.0/perl-5.22.1.tar.gz
tar xvfz perl-5.22.1.tar.gz
cd perl-5.22.1 && ./Configure -Duseithreads -des && make && make test && make install_**

It will put the new perl bin in /usr/local/bin/perl
so you need to call perl with **/usr/local/bin/perl** and call cpan with **/usr/local/bin/cpan**

after that i had problems with installing IO::Socket::SSL. It would generate the following error: 

**_make: *** [blib/arch/auto/Net/SSLeay/SSLeay.so] Error 1_**

So I had to recompile openssl with switch -fPIC 

./configure -fPIC

https://geeksww.com/tutorials/libraries/openssl/installation/installing_openssl_on_ubuntu_linux.php


After that i received an error with libXML:

**/usr/local/bin/cpan XML::LibXML**

**_make: *** [blib/arch/auto/XML/LibXML/LibXML.so] Error 1_**

So, i had to recompile zlib with switch --shared

./configure  --shared

https://geeksww.com/tutorials/libraries/zlib/installation/installing_zlib_on_ubuntu_linux.php


I think i covered all but im not sure :D

**Comment by ERR1R** [comment]
btw, if its not returning a msg id, is header checking still working? Because that is the reason i want to switch to newsup.. all the other posters are lacking this feature.

How is this working with newsup? Do you need an nzb for header checking?

**Comment by demanuel** [comment]
No. This last version, if the server doesn't yield the message ID the headerchecking isn't working.

Please use the version https://github.com/demanuel/NewsUP/tree/8659ae2ff8188036f5fe08716e7fac10c982df1d

That version is 100% functional. Just replace the newsup.pl file. Until i fix the issue, that version is the recommended one to use.

Regarding your troubles to use perl 5.022, you can downgrade the version newsup requires by changing the [second line](https://github.com/demanuel/NewsUP/blob/master/newsup.pl#L2). I'm requesting version 5.022 because that version brings a lot of speed improvements in their internals. You should be fine using any version above 5.014

**Comment by peaveyman** [comment]
I would replace the newsup in /usr/bin correct?

**Comment by demanuel** [comment]
yes

**Comment by peaveyman** [comment]
Or rather usr/bin/vendor-perl

**Comment by demanuel** [comment]
Please replace 

/usr/bin/vendor_perl/newsup

It's where the pkgbuild installs it

**Comment by peaveyman** [comment]
Is there a fix for this issue in the future?

**Comment by demanuel** [comment]
It's on my TODO list, but unfortunately I don't  any idea when this is going to be fixed, as big changes are required.

**Comment by peaveyman** [comment]
No worries. Just was wondering about it. It works great as is.

**Comment by demanuel** [comment]
@ERR1R 
@peaveyman 

Please check this version: https://github.com/demanuel/NewsUP/tree/v4

Please note that i changed a lot of stuff so the configuration file needs to be modified.

**Comment by peaveyman** [comment]
How does the configuration file need to be modified?

**Comment by demanuel** [comment]
@peaveyman updated the pkgbuild in the branch.

The best option for the conf file is to copy the one distributed with the files and fill it again.

You can specified a configuration file with --configuration ....
If you don't specify, it will use the ~/.conf/newsup.conf

**Comment by peaveyman** [comment]
I have it installed but am getting this error:

The `temp_folder` doesn't exist! Please create it. at /usr/bin/vendor_perl/newsup line 37

**Comment by demanuel** [comment]
on the config file;
[options]
temp_folder=/point/this/to/a/valid/folder

**Comment by peaveyman** [comment]
Tried that, doesn't work. Same error.

**Comment by demanuel** [comment]
If you're getting that error than the folder you defined doesn't exist:

https://github.com/demanuel/NewsUP/blob/v4/lib/NewsUP/Utils.pm#L276

Are you using the right config file?

**Comment by peaveyman** [comment]
I have the config file in ~/.config

**Comment by demanuel** [comment]
Are you sure you don't have any typo in the config file? It's temp_folder not tmp_folder.

Otherwise i have no clue what's going on, as i'm unable to reproduce it. :-(

**Comment by demanuel** [comment]
Does the folder: "/home/gary/tmp" exists?

The files are copied to that location, and then all the operations are done over those files. It's a safer way to ensure that i don't screw up the uploading files.

**Comment by peaveyman** [comment]
Yes, that folder exists. I created it myself when I first got the error.

**Comment by demanuel** [comment]
if you do this on the command line:

    perl -e 'print "NO\n" if !-e "/home/gary/tmp"'

what do you get?

**Comment by peaveyman** [comment]
Nothing, no error, nothing. Just goes back to the prompt.

**Comment by demanuel** [comment]
Please update the package. 

I added a --debug switch so we can see what value is using for the options.

after you update just:

    newsup --debug

**Comment by peaveyman** [comment]
Ok, here is what I get with the debug:

Using the options: 
        AUTH_PASS: ******
        PAR2_SETTINGS: c -s768000 -r15
        SERVER_PORT: 563
        HEADERCHECK_RETRIES: 3
        PROGRESSBAR_SIZE: 16
        DEBUG: 1
        SERVER: secure.news.thecubenet.com
        HEADERCHECK_AUTH_PASS: ******
        NZB_SAVE_PATH: /data/uploads/
        HEADERS: HASH(0x55e56f033d28)
        TEMP_FOLDER: /data/tmp
        UPLOAD_NZB: 1 # if the nzb is also uploaded
        SPLIT_SIZE: 50 # The size of the split rars
        CONNECTIONS: 5
        HEADERCHECK: 1
        SPLIT_PATTERN: *7z *[0-9][0-9][0-9]
        GROUPS:
                alt.binaries.sounds.lossless.country
                 alt.binaries.sounds.flac
                 alt.binaries.sounds.lossless
        GENERATE_IDS: 0 # To generate random ids or to use the returned ones from the server. Note: Some servers don't return anything. In that case change this option to 1
        HEADERCHECK_SERVER: secure.news.thecubenet.co
        PAR2_PATH: par2 # full path to par2 executable
        PAR2: 0 # enable or disable the par'ing
        NZB_FILE: Daryle_Singletary-Daryle_Singletary-Flac-1995.nzb
        HEADERCHECK_SERVER_PORT: 563
        COMMENTS:
                Daryle_Singletary-Daryle_Singletary-Flac-1995
        PAR2_RENAME_SETTINGS: c -s768000 -r0
        METADATA: HASH(0x55e56f033440)
        HEADERCHECK_CONNECTIONS: 3
        FILES:
                Daryle_Singletary-Daryle_Singletary-Flac-1995
        HEADERCHECK_AUTH_USER: ******
        AUTH_USER: ******
        UPLOADER: ****** <News@Newsconnection.local>
        TLS: 1
        UPLOAD_SIZE: 768000
The `temp_folder` doesn't exist! Please create it. at /usr/bin/vendor_perl/newsup line 37.

**Comment by peaveyman** [comment]
How do I disable the par and rar section of the config file?

**Comment by demanuel** [comment]
From what i can see it's using the value /data/tmp for the temp_folder.
Please check if you don't have the option defined twice or if your're using the right file (it will use the ~/.config/newsup.conf )

To disable the par and rar just use:

    [options]
    splitnpar = 0

**Comment by peaveyman** [comment]
I have this: 

[options]
rarnpar= 0

Should it be: 

[options]
rarnpar = 0

**Comment by demanuel** [comment]
That was on the previous version. Now this version i tried to be more application agnostic (You can use either with rar or 7zip), since rarnpar would assume that it would use only rar, i changed the option name to splitnpar.

**Comment by peaveyman** [comment]
I think there is a typo in the config file. I copied the new one from /etc/newsup and it has this:

[options]
rarnpar= 0

I changed it to splitnpar = 0 and it seems to be uploading. But on the command line I get this: Not enough command line arguments.

**Comment by peaveyman** [comment]
The header check didn't work with the config having enabled= 1. Trying it with enabled = 1 now. It uploaded just fine but I received this at the end:

Can't locate object method "add_child" via package "XML::LibXML::Element" at /usr/share/perl5/vendor_perl/NewsUP/Utils.pm line 192.

**Comment by peaveyman** [comment]
The header check didn't work with enabled = 1 either.

**Comment by demanuel** [comment]
i just updated the code, that addresses the XML and the rarnpar config typo.

do you have that option on 

    [headerCheck]
    enabled = 1
    retries = 3

**Comment by peaveyman** [comment]
Yes, the option is on. I have noticed in several places in the config that some have something like this: 

enabled = 1 and some have enabled= 1

Does it make a difference? Even the splitnpar you posted on this thread shows to have a space before the = but in the config there is no space.

**Comment by demanuel** [comment]
There's no difference. It can be with or without space.

Can you paste again the --debug output?

**Comment by peaveyman** [comment]
I have a script that runs the par operation and then posts using newsup. Do you want the debug from newsup there or just newsup --debug?

**Comment by demanuel** [comment]
If you can provide both the better :-)

**Comment by peaveyman** [comment]
Ok, here is newsup --debug:

newsup --debug
Using the options: 
        SERVER: secure.news.thecubenet.com
        HEADERCHECK_CONNECTIONS: 3
        HEADERCHECK: 1
        HEADERCHECK_SERVER: secure.news.thecubenet.co
        PAR2_SETTINGS: c -s768000 -r15
        TEMP_FOLDER: /home/gary/tmp
        AUTH_PASS: ****
        SERVER_PORT: 563
        NZB_SAVE_PATH: /data/uploads/
        GENERATE_IDS: 0 # To generate random ids or to use the returned ones from the server. Note: Some servers don't return anything. In that case change this option to 1
        HEADERCHECK_RETRIES: 3
        HEADERCHECK_SERVER_PORT: 563
        PAR2: 0 # enable or disable the par'ing
        AUTH_USER: ****
        PAR2_PATH: par2 # full path to par2 executable
        PAR2_RENAME_SETTINGS: c -s768000 -r0
        TLS: 1
        SPLIT_PATTERN: *7z *[0-9][0-9][0-9]
        UPLOAD_NZB: 1 # if the nzb is also uploaded
        DEBUG: 1
        CONNECTIONS: 5
        HEADERCHECK_AUTH_PASS: ***
        PROGRESSBAR_SIZE: 16
        HEADERS: HASH(0x55d8e6247608)
        SPLIT_SIZE: 50 # The size of the split rars
        METADATA: HASH(0x55d8e6247b78)
        UPLOAD_SIZE: 768000
        UPLOADER: *** <News@Newsconnection.local>
        GROUPS:
                alt.binaries.sounds.lossless.country
                 alt.binaries.sounds.flac
                 alt.binaries.sounds.lossless
        HEADERCHECK_AUTH_USER: ***

**Comment by peaveyman** [comment]
It's not finished uploading but here is the debug from the start of the upload:

Using the options: 
        METADATA: HASH(0x55ac70979da0)
        SPLIT_SIZE: 50 # The size of the split rars
        PAR2_PATH: par2 # full path to par2 executable
        HEADERS: HASH(0x55ac709749b0)
        SERVER_PORT: 563
        NZB_SAVE_PATH: /data/uploads/
        AUTH_PASS: ****
        HEADERCHECK_SERVER_PORT: 563
        UPLOAD_NZB: 1 # if the nzb is also uploaded
        GROUPS:
                alt.binaries.sounds.lossless.country
                 alt.binaries.sounds.flac
                 alt.binaries.sounds.lossless
        TLS: 1
        SPLIT_PATTERN: *7z *[0-9][0-9][0-9]
        PAR2_SETTINGS: c -s768000 -r15
        UPLOADER: **** <News@Newsconnection.local>
        GENERATE_IDS: 0 # To generate random ids or to use the returned ones from the server. Note: Some servers don't return anything. In that case change this option to 1
        CONNECTIONS: 5
        HEADERCHECK_CONNECTIONS: 3
        HEADERCHECK: 1
        COMMENTS:
                Daryle_Singletary-Now_and_Again-Flac-2000
        SERVER: secure.news.thecubenet.com
        PAR2: 0 # enable or disable the par'ing
        HEADERCHECK_AUTH_PASS: ****
        NZB_FILE: Daryle_Singletary-Now_and_Again-Flac-2000.nzb
        HEADERCHECK_AUTH_USER: ****
        HEADERCHECK_SERVER: secure.news.thecubenet.co
        AUTH_USER: ****
        PAR2_RENAME_SETTINGS: c -s768000 -r0
        UPLOAD_SIZE: 768000
        TEMP_FOLDER: /home/gary/tmp
        HEADERCHECK_RETRIES: 3
        FILES:
                Daryle_Singletary-Now_and_Again-Flac-2000
        PROGRESSBAR_SIZE: 16
        DEBUG: 1
Not enough command line arguments.

**Comment by peaveyman** [comment]
The header check still didn't work.

**Comment by peaveyman** [comment]
When I update the app I can just run makepkg -si in the directory where the PKGBUILD resides can't I?

**Comment by demanuel** [comment]
The "Not enough command line arguments" line is from the par2 utilitiy.

**Comment by peaveyman** [comment]
Is there supposed to be a progress indicator for the header check?

**Comment by demanuel** [comment]
No. Only when uploading segments.

**Comment by demanuel** [comment]
@peaveyman 
> When I update the app I can just run makepkg -si in the directory where the PKGBUILD resides can't I?

Yes.

Let me know if you have questions regarding V4.

**Comment by demanuel** [comment]
@peaveyman let me know if i can close this issue.

Anything not related with this issue, please open a new issue.

**Comment by demanuel** [comment]
This is solved in the latest version.
Please note that a lot has changed so you'll need a new config file.
